lib/auth/auth_with_roles.go

```diff
@@ -2601,6 +2611,132 @@ func (a *ServerWithRoles) GenerateUserSingleUseCerts(ctx context.Context) (proto
 	return nil, trace.NotImplemented("bug: GenerateUserSingleUseCerts must not be called on auth.ServerWithRoles")
 }
 
+func (a *ServerWithRoles) IsMFARequired(ctx context.Context, req *proto.IsMFARequiredRequest) (*proto.IsMFARequiredResponse, error) {
+	if err := a.currentUserAction(req.Username); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var noMFAAccessErr, notFoundErr error
+	switch t := req.Target.(type) {
```

Comment:
More of a meta comment, but don't we want to keep the ACL layer slim without much business logic? Any downsides of moving this switch downstream, e.g. to an auth server method and invoke it here similar to how other methods here do? You can pass checker to it too.

Reply:
All of this code felt like pure authz logic, which is why I put it here.
But I think you're right, moved all of this code into `auth.Server`.